examples/music_drop_fade.py

```diff
@@ -0,0 +1,206 @@
+
+    It looks in the file directory and its data subdirectory
+    """
+    if filename.rpartition(".")[2] not in file_types:
+        print("{} not added to file list".format(filename))
+        print("only these files types are allowed: ", file_types)
+        return False
+    elif os.path.exists(filename):
+        file_list.append(filename)
+    elif os.path.exists(os.path.join(main_dir, filename)):
+        file_list.append(os.path.join(main_dir, filename))
+    elif os.path.exists(os.path.join(data_dir, filename)):
+        file_list.append(os.path.join(data_dir, filename))
+    else:
+        print("file not found")
+        return False
+    print("{} added to file list".format(filename))
+    return True
+
+
+def play_file(filename):
+    """
+    This function will call add_file and play it if successful
+    The music will fade in over the first 4 seconds
+    set_endevent is used to post a MUSIC_DONE event when the song finishes
+    The main loop will call play_next() when the MUSIC_DONE event is received
+
+    """
+    if add_file(filename):
+        try:  # we must do this in case the file is not a valid audio file
+            pg.mixer.music.load(file_list[-1])
+        except pg.error as e:
+            print(e)  # print a description such as 'Not an Ogg Vorbis audio stream'
+            if filename in file_list:
+                file_list.remove(filename)
+                print("{} removed from file list".format(filename))
+            return
+        pg.mixer.music.play()  # fade_ms=4000)
+        pg.mixer.music.set_volume(volume)
+        pg.mixer.music.set_endevent(MUSIC_DONE)
+
+
+def play_next():
+    """
+    This function is will play the next song in file_list
+    It uses pop(0) to get the next song and then appends it to the end of the list
+    The song will fade in over the first 4 seconds
+    """
+    if len(file_list) > 1:
+        nxt = file_list.pop(0)
+
+        try:
+            pg.mixer.music.load(nxt)
+        except pg.error as e:
+            print(e)
+            print("{} removed from file list".format(nxt))
+
+        #    print('{} is not a valid music file'.format(nxt))
+        #    return
+
+        file_list.append(nxt)
+        print("starting next song: ", nxt)
+    pg.mixer.music.play()  # fade_ms=4000)
+    pg.mixer.music.set_volume(volume)
+    pg.mixer.music.set_endevent(MUSIC_DONE)
+
+
+def draw_line(text, y=0):
+    """
+    Draws a line of text onto the display surface
+    The text will be centered horizontally at the given y postition
+    The text's height is added to y and returned to the caller
+    """
+    screen = pg.display.get_surface()
+    surf = font.render(text, 1, (255, 255, 255))
+    y += surf.get_height()
+    x = (screen.get_width() - surf.get_width()) / 2
+    screen.blit(surf, (x, y))
+    return y
+
+
+MUSIC_DONE = pg.USEREVENT + 1  # create a user event called MUSIC_DOWN in the first slot
```

Comment:
We recently added: `pygame.event.custom_type()` https://www.pygame.org/docs/ref/event.html#pygame.event.custom_type

```python
MUSIC_DONE = pg.event.custom_type()
```